### PR TITLE
Refactor datasets interface and add support for a drugcomb dataset

### PR DESCRIPTION
**Pull request by thomasgaudelet**
## What is the goal of this PR?

This PR focuses on the suite of datasets part of the library.

It first refactors the existing datasets interface, breaking down the large files into smaller ones to facilitate maintenance.
Second, it adds DrugCombDataset object that interface with the drugcomb database of drug combinations.

## What are the changes implemented in this PR?
- breaks down `classification.py` and `regression.py` into multiple files to facilitate maintenance. 
- abstracts away some common functionalities
- expose the random seed parameter for all datasets to ensure reproducibility
- adds DrugCombDataset

### CODE REVIEW

File: pyrelational/datasets/classification/mnist.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+from torch import Tensor
+from torch.utils.data import ConcatDataset
+from torchvision import datasets, transforms
+
+from pyrelational.datasets.base import BaseDataset
+
+from .utils import create_splits
+
+
+class FashionMNIST(BaseDataset):
```
Comment (paulmorio) [inline pyrelational/datasets/classification/mnist.py:13]: Super minor comment: I think it should be called fashion_mnist.py

File: pyrelational/datasets/classification/utils.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    """
+    remapped_labels: Tensor = torch_class_array.unique(return_inverse=True)[1]
+    return remapped_labels
+
+
+def create_splits(x: Tensor, y: Tensor, n_splits: int, random_seed: int) -> List[NDArray[np.int_]]:
+    """
+    Create stratified k-fold splits for the dataset using the dataset's features and labels.
+    """
+    skf = StratifiedKFold(n_splits=n_splits, random_state=random_seed, shuffle=True)
```
Comment (paulmorio) [inline pyrelational/datasets/classification/utils.py:30]: I am curious what happens when using StratifiedKFold on regression datasets?

**Review by paulmorio — COMMENTED** [review]
Looks great, thanks for all the additions, just a quick question about what happens on the stratified K-fold splits on regression datasets.

**Review by paulmorio — APPROVED** [review]
LGTM!
